src/common/QueryHandler.ts

```diff
@@ -191,11 +210,46 @@ const pierceHandler = createInternalQueryHandler({
   },
 });
 
-const builtInHandlers = new Map([
-  ['aria', ariaHandler],
-  ['pierce', pierceHandler],
+const xpathHandler = internalizeCustomQueryHandler({
+  queryOne: (element, selector) => {
+    const doc = element.ownerDocument || document;
+    const iterator = doc.evaluate(
+      selector,
+      element,
+      null,
+      XPathResult.ORDERED_NODE_ITERATOR_TYPE
+    );
+    return iterator.iterateNext();
+  },
+
+  queryAll: (element, selector) => {
+    const doc = element.ownerDocument || document;
+    const iterator = doc.evaluate(
+      selector,
+      element,
+      null,
+      XPathResult.ORDERED_NODE_ITERATOR_TYPE
+    );
+    const array: Node[] = [];
+    let item;
+    while ((item = iterator.iterateNext())) {
+      array.push(item);
+    }
+    return array;
+  },
+});
+
+interface InternalQueryHandlerInfo {
+  test: RegExp;
+  handler: InternalQueryHandler;
+}
+
+const INTERNAL_QUERY_HANDLERS = new Map<string, InternalQueryHandlerInfo>([
+  ['aria', {test: /$aria\//g, handler: ariaHandler}],
+  ['pierce', {test: /$pierce\//g, handler: pierceHandler}],
+  ['xpath', {test: /$\.?\/\//, handler: xpathHandler}],
```

Comment:
will it accept `/`?  I think it's a valid XPATH

Reply:
I think we should also support `(//h1)[1]/text()` and  perhaps `count(//*)`?

Reply:
so it does not look like we can test it by a regex?

Reply:
`(//h1)[1]/text()` is the same as `//h1[1]/text()`.

I don't think we should support XPaths that give non-Node return values. The reason being we cannot detect (without going down a parsing rabbit hole) whether the return value is not a Node which is required for the correct `XPathResult` enum.

Reply:
So why should not we support `(//h1)[1]/text()` it is a valid selector, right? I suggest that we prefix the xpath selectors with `xpath/` that allows defining any XPath selectors the user might want but automically rewrite `.//`, `//`, and `/` to have the prefix `xpath/`. WDYT?

Reply:
sg